zerver/worker/queue_processors.py

```diff
@@ -527,3 +535,37 @@ def consume(self, event: Mapping[str, Any]) -> None:
                 (stream, recipient, sub) = access_stream_by_id(user_profile, stream_id,
                                                                require_active=False)
                 do_mark_stream_messages_as_read(user_profile, stream)
+
+
+@assign_queue('export_data')
+class DataExporter(QueueProcessingWorker):
+    def consume(self, email: str) -> None:
+        output_dir = tempfile.mkdtemp(prefix="/tmp/zulip-export-")
+        if os.path.exists(output_dir):
+            shutil.rmtree(output_dir)
+        os.makedirs(output_dir)
+
+        user_profile = get_user_profile_by_email(email)
+
+        # FIXME: Can we check if an export is currently running, or has been
+        # started in the past hour, and deny exports?
+        do_export_user(user_profile, output_dir)
+        print("Finished exporting to %s; tarring" % (output_dir,))
+        tarball_path = output_dir.rstrip('/') + '.tar.gz'
+        subprocess.check_call(["tar", "--strip-components=1", "-czf", tarball_path, output_dir])
+        print("Tarball written to %s" % (tarball_path,))
+
+        # Upload
+        with open(tarball_path, 'rb') as f:
+            content_type = mimetypes.guess_type(tarball_path)[0]
+            size = getsize(tarball_path)
+            name = basename(tarball_path)
+            print(content_type, name, size)
+            upload_path = upload_message_image(name, size, content_type, f.read(), user_profile)
```

Comment:
Hmm, is that function really called `upload_message_image`?  Sigh, that name is super confusing.  Can you do a separate PR (or commit at the base of this PR) to rename that to `upload_message_file`?  That seems like a clear thing to merge first.  Be sure to use good commit discipline for that refactor; check out the Zulip commit message guidelines for more details: https://zulip.readthedocs.io/en/latest/contributing/version-control.html